src/main/java/it/mulders/mcs/cli/Cli.java

```diff
@@ -65,13 +67,27 @@ public class SearchCommand implements Callable<Integer> {
         )
         private Integer limit;
 
+        @CommandLine.Option(
+                names = { "-f", "--format" },
+                description = """
+                        Show result in <type> format
+                        Supported types are:
+                          maven, gradle, gradle-short, gradle-kotlin, sbt, ivy, grape, leiningen, buildr
+                        """,
+                paramLabel = "<type>"
+        )
+        private String responseFormat;
+
         @Override
         public Integer call() {
             var combinedQuery = String.join(" ", query);
             System.out.printf("Searching for %s...%n", combinedQuery);
             var searchQuery = SearchQuery.search(combinedQuery)
                     .withLimit(this.limit)
                     .build();
+
+            CoordinatePrinter coordinatePrinter = FormatType.providePrinter(responseFormat);
+            searchCommandHandler.setCoordinatePrinter(coordinatePrinter);
```

Comment:
I would prefer the `SearchCommandHandler` class to stay immutable. I think the only reason we cannot do that right now is because it is injected from the `App` class, right?

Would it be possible to only construct it once we know the `CoordinatePrinter` implementation, so we can keep it immutable?

Reply:
`DelegatingOutputPrinter` is immutable again. `SearchCommandHandler` is constructed locally, after `CoordinatePrinter` is determined from CLI parameters.  
However, this change broke `SearchCommandHandler` injecting as a `Cli` constructor parameter.
That is the reason for `MockedConstruction` usage in `CliTest`.
I'm not sure whether this solution is appropriate.  
I also haven't found any elegant solution for testing, which `CoordinatePrinter` implementation will be injected into `DelegatingOutputPrinter` within `CliTest`.